### PR TITLE
feat: overhaul collection view

### LINKED ISSUE #554: Collection View Refactor
Improvements for the Collection view will allow the user to quickly understand the status of datasets in their Qri collection, find datasets to work with, and perform dataset-level actions (publish/unpublish, remove, etc).  The collection view is "home base" and is where users will take control of their datasets: pushing, pulling, removing, finding, and just understanding what they have.

This work assumes refactoring of the top-level layout of Qri Desktop, with a fixed-height header running the full width of the content pane.  (The current layout uses split sidebar/content on all views without a unified header)

Major Functionality
- List of Datasets - The list shows all datasets in the Qri collection, and is sortable by clicking the 

- Filtering -  The user can filter by string match or date range of last update (we can add more filters later, these are two simple ones to start with) Filter pane can get more complex as users demand more ways to navigate large collections.  One idea is to have a collapsible advanced filter section.

- Dataset status glyphs - Status glyphs provide a quick visual indicator of published/unpublished, file system integration, and sync.   

- Dataset actions - An ellipses icon at the end of each row shows a context menu with dataset-level actions (remove, duplicated, publish, etc)

- Multiselect - User can select multiple datasets using checkboxes and perform bulk actions.

TODO:
- Indicate how many datasets are currently selected, visually highlight selected datasets in addition to the checkbox 
- Warning UI for remove/publish - These currently exist in the app, but would need to be modified to handle multi-select actions
- Think about keyboard-only navigation and actions
- Syncing UI - clearly show that Qri is pulling new data
- Visual indication that a dataset has "new" data that the user hasn't seen yet

Qri_Desktop_0_4_0_-_v4_sketch

### LINKED ISSUE #608: refactor(TopNavbar): dataset action buttons should be moved to the TopNavbar
The current `TopNavbar` goes:

- arrows
- location
- title

let's refactor so it looks closer to this:
94437527-e0d33a80-016b-11eb-97cb-83eab172399b

- arrows
- title / action buttons
- location

### PR DESCRIPTION
**Pull request by b5**
closes #554, closes #608

http://www.giphy.com/gifs/9hpV3Qyp84e4DEvRnC

Lots of work left to do, and styling leaves a few things to be desired, but I think this is a good start that we can build on

### CODE REVIEW

File: app/components/item/VersionInfoItem.tsx [hunk 1]
```diff
@@ -1,38 +1,55 @@
...
+      <td className='icons'>
+        <ExternalLink href={`${QRI_CLOUD_URL}/${username}/${name}`} className="dataset-link" tooltip="Published">
+          <Icon icon="publish" size="sm" color={published ? 'dark' : 'medium'}/>
+        </ExternalLink>
+        {onClickFolder && <a onClick={(e: React.MouseEvent) => onClickFolder(data, e)} className="dataset-link" data-tip="Linked to filesystem">
+          <Icon icon="openInFinder" size="sm" color={fsiPath ? 'dark' : 'medium'}/>
+        </a>}
+        {/* <button onClick={() => updateDataset(username, name)} className="dataset-button" data-tip="Updates available">
+          <Icon icon="sync" size="sm" color={updatesAvailable ? 'dark' : 'medium'}/>
+        </button> */}
```
Comment (b5) [inline app/components/item/VersionInfoItem.tsx:50]: @uhLeeshUh I've inlined the `DatasetStatus` component here, seemed to make more sense directly inlined, but if we need to use these icons elsewhere, I'd be all for breaking it back out

File: app/components/collection/collectionHome/DatasetCollection.tsx [hunk 2]
```diff
@@ -63,29 +59,15 @@ export const CollectionHomeMainContentComponent: React.FunctionComponent<Collect
...
-        <HeaderColumnButton
-          icon={faDownload}
-          id='pull-dataset'
-          label='Add existing Dataset'
-          onClick={() => { setModal({ type: ModalType.PullDataset }) }}
-        />
-      </div>
       <div className='main-content-flex'>
-        <DatasetList />
+        <DatasetList showFSI={true} />
```
Comment (uhLeeshUh) [inline app/components/collection/collectionHome/DatasetCollection.tsx:63]: Very small nit here but can drop the `={true}` for a truthy prop like this

**Review by ramfox — APPROVED** [review]
This looks great! So exciting!

LGTM

**Comment by chriswhong** [comment]
Looks great!  Some requests, which we can always do later:

- format row counts (159319 -> 159.3k)
- tooltips on status icons
- column sorting (click column title to sort, show sort direction indicator ꜛꜜ
- tooltip on `updated` to show actual timestamp, like we do in cloud (13 days ago -> Thu, Aug 15, 2020 10:15am)
- text wraps when Desktop window is shortened, recommend making the name column flex/show elipses and keeping the other columns static (see image below)
- "My Datasets" vs "All Local Datasets" is hard to follow, makes me think My Datasets are not local, or somehow I am not getting the idea that My Datasets is included in All Local Datasets.  Maybe My Datasets/All Datasets.  
- Consider adding a number next to the tab label like "My Datasets (7)" | "All Datasets (28)"
- When dataset actions are shown, "Showing XX datasets" is shifted to the right.


Qri_Desktop

File: app/components/collection/collectionHome/DatasetCollection.tsx (hunk 2 above)
Comment (b5) [reply app/components/collection/collectionHome/DatasetCollection.tsx:63]: very good nit, will fix!

**Comment by b5** [comment]
such great feedback @chriswhong. I've addressed the following:

* [x] format row counts (159319 -> 159.3k)
* [x] tooltips on status icons
* [x] tooltip on updated to show actual timestamp, like we do in cloud (13 days ago -> Thu, Aug 15, 2020 10:15am)
* [x] "My Datasets" vs "All Local Datasets" is hard to follow, makes me think My Datasets are not local, or somehow I am not getting the idea that My Datasets is included in All Local Datasets. Maybe My Datasets/All Datasets.
* [x] Consider adding a number next to the tab label like "My Datasets (7)" | "All Datasets (28)"
* [x] When dataset actions are shown, "Showing XX datasets" is shifted to the right.

I'll open issues for these:
* column sorting (click column title to sort, show sort direction indicator ꜛꜜ
* text wraps when Desktop window is shortened, recommend making the name column flex/show elipses and keeping the other columns static (see image below)
